data_science/01.00-addition-laws.py

```diff
@@ -0,0 +1,416 @@
+# /// script
+# requires-python = ">=3.10"
+# dependencies = [
+#     "marimo",
+#     "matplotlib",
+#     "matplotlib-venn"
+# ]
+# ///
+
+import marimo
+
+__generated_with = "0.10.19"
+app = marimo.App()
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        """
+        # 🎲 Addition Laws of Probability
+
+        Welcome to the world of combining probabilities! In this notebook, we'll learn how to:
+
+        - Add probabilities of simple events
+
+        - Work with mutually exclusive events
+
+        - Use the general addition rule
+
+        Let's make probability addition fun and visual! 🎯
+        """
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        """
+        ## 🎯 Simple Events
+
+        Simple events are basic outcomes that don't overlap. Think of rolling a die:
+
+        - Rolling a 1 is a simple event
+
+        - Rolling a 2 is another simple event
+
+        - They can't happen at the same time!
+        """
+    )
+    return
+
+
+@app.cell
+def _(die_events):
+    die_events
+    return
+
+
+@app.cell
+def _(mo):
+    # Interactive die probability calculator
+    die_events = mo.ui.multiselect(
+        options=["1", "2", "3", "4", "5", "6"],
+        value=["1", "2"],
+        label="Select die numbers"
+    )
+    return (die_events,)
+
+
+@app.cell
+def _(die_events, mo, plt):
+    selected = [int(x) for x in die_events.value]
+    die_prob = len(selected) / 6
+
+    # Create simple bar plot
+    plt.figure(figsize=(8, 4))
+    plt.bar(['Selected', 'Others'], 
+            [die_prob, 1-die_prob],
+            color=['#2ecc71', '#e74c3c'])
+    plt.title('Probability Distribution')
+    plt.ylabel('Probability')
+    plt.ylim(0, 1)
+
+    die_result = mo.md(f"""
+    ### Probability Analysis
+
+    Selected numbers: {', '.join(die_events.value)}
+
+    P(selected) = {die_prob:.3f}
+
+    This is a simple addition because each number can only come up once!
```

Comment:
Here, "addition" is referred to but it hasn't been defined yet. What is being added? I think I know, but it is worth being more explicit, even writing out the latex for the corresponding event.

`P(\{1\} \cup \{2\} ...) = P(1) + P(2) + ... = ...`